x-pack/plugins/security_solution/server/lib/detection_engine/signals/utils.ts

```diff
@@ -606,7 +606,15 @@ export const createSearchAfterReturnTypeFromResponse = ({
   timestampOverride: TimestampOverrideOrUndefined;
 }): SearchAfterAndBulkCreateReturnType => {
   return createSearchAfterReturnType({
-    success: searchResult._shards.failed === 0,
+    success:
+      searchResult._shards.failed === 0 ||
+      searchResult?._shards?.failures?.every((failure) => {
```

Comment:
This is a bit of funny logic. Are you getting paranoid everywhere and using the optional chaining, `?` in most places right now? ;-) Or are our types wrong?

Because above this we have:

```ts
searchResult._shards.failed === 0 ||
```

which would blow up if `searchResult` or `_shards` is `undefined` but then after that or statement you have code with optional chaining that says it could be undefined:

```ts
searchResult?._shards?.failures
```

Did you just want something like this:

```ts
searchResult._shards.failures?.every
```

I would be careful about overuse of the optional chaining operator and only use it where it is needed.

Reply:
ah yeah good catch. The last one `searchResult._shards.failures?.every` is all I need. Thanks!